<?xml version="1.0"?>
<!-- 
    The UGV is only simulated and shown in RViz to help tuning the PID Controller
-->
<launch>
<!--
    Display the UGV first
-->
<!---->

    <arg name="simulated_robot" default="true" />
    <arg name="control_rate" default="50" />
    <arg name="publish_tf" default="true" />


    <arg
        name="model" />
    <param
        name="robot_description"

        textfile="$(find bunker_pro_description)/urdf/bunker_pro.urdf.xacro" />

<!--
    <include file="$(find chsweld_support)/launch/chsweld.launch" />
-->
    <!-- There is no joints to move therefore do not use gui -->

    <node
        name="joint_state_publisher_gui"
        pkg="joint_state_publisher_gui"
        type="joint_state_publisher_gui" />
    <node
        name="robot_state_publisher"
        pkg="robot_state_publisher"
        type="robot_state_publisher" />
    <node
        name="rviz"
        pkg="rviz"
        type="rviz"
        args="-d $(find bunker_pro_description)/rviz/bunker_pro.rviz" />

<!--
    Start the Teleop Keyboard

    <arg name="repeat_rate" value="50.0"/>
    <node name="teleop_keybord" pkg="teleop_twist_keyboard" type="teleop_twist_keyboard.py" output="screen">
        <param name="repeat_rate" value="$(arg repeat_rate)"/>
    </node>

    <arg name="simulated_robot" default="true" />
    <arg name="control_rate" default="50" />
    <arg name="publish_tf" default="true" />
-->
<!--
    Bring up the UGV
-->
    <node name="bunker_pro_base_sim_node" pkg="ugv_simulation" type="bunker_pro_base_sim_node" output="screen">
        <param name="simulated_robot" type="bool" value="$(arg simulated_robot)" />
        <param name="control_rate" type="int" value="$(arg control_rate)" />
        <param name="publish_tf" type="bool" value="$(arg publish_tf)" />

        <param name="odom_frame" type="string" value="odom" />
        <param name="base_frame" type="string" value="dummy_link" />
    </node>
</launch>